solana/programs/example-native-token-transfers/src/instructions/release_inbound.rs

```diff
@@ -138,22 +139,22 @@ pub fn release_inbound_unlock(
     assert!(inbox_item.release_status == ReleaseStatus::Released);
     match ctx.accounts.common.config.mode {
         Mode::Burning => Err(NTTError::InvalidMode.into()),
-        Mode::Locking => token_interface::transfer_checked(
-            CpiContext::new_with_signer(
-                ctx.accounts.common.token_program.to_account_info(),
-                token_interface::TransferChecked {
-                    from: ctx.accounts.custody.to_account_info(),
-                    to: ctx.accounts.common.recipient.to_account_info(),
-                    authority: ctx.accounts.common.token_authority.clone(),
-                    mint: ctx.accounts.common.mint.to_account_info(),
-                },
+        Mode::Locking => {
+            onchain::invoke_transfer_checked(
+                &ctx.accounts.common.token_program.key(),
+                ctx.accounts.custody.to_account_info(),
+                ctx.accounts.common.mint.to_account_info(),
+                ctx.accounts.common.recipient.to_account_info(),
+                ctx.accounts.common.token_authority.clone(),
+                ctx.remaining_accounts,
```

Comment:
So I see this is how the API is intended to be used - but, what accounts are we intending to pass along here? Multi-sigs?

`remaining_accounts` is considered to be semi-dangerous because it jettisons Anchor's usual checks on these accounts. If we use it we should be clear about why and why it's safe.

See:
https://docs.rs/anchor-lang/latest/anchor_lang/context/struct.Context.html#structfield.remaining_accounts
https://www.zellic.io/blog/the-vulnerabilities-youll-write-with-anchor/#misuse-of-anchors-ctxremaining_accounts

Reply:
`remaining_accounts` is dangerous when used directly because no account checks have been made by our program. In this case, these accounts are just passed on directly to the token2022 program, which in turn will pass them to the transfer hook, which is responsible for validating all the usual things about it (in practice the transfer hook will be an anchor program).
So there's no real security impact, as we're not using those accounts directly.

The accounts that are actually passed in are computed by the client here https://github.com/wormhole-foundation/example-native-token-transfers/blob/29fb437c3dbae0e5f018731e655f54f73660a1b5/solana/ts/sdk/ntt.ts#L438-L445 by consulting the transfer hook program's `extra-accounts-meta` PDA. The `invoke_transfer_checked` function then does some verificaiton on these accounts, and ultimately passes them to the token2022 program.